### ISSUE TITLE
Add internet permission on Android only for debug build type

### ISSUE THREAD

**Issue by zimmi** [flutter/flutter#20789]
When creating a new app through `flutter create`, `<uses-permission android:name="android.permission.INTERNET"/>` is added to the main `AndroidManifest.xml`. 

That's inconvenient for apps that don't need internet access because it [adds a manual step to the release process](https://flutter.io/android-release/#review-the-app-manifest).
It's also easy to forget to add it back after the release as seen here: https://stackoverflow.com/questions/51751873/flutter-app-does-not-start-in-debug-mode/51751874#51751874

A quick implementation idea:
1. Add the folder `debug` next to `main`
1. Create a new `AndroidManifest.xml` inside `debug` with the permission snippet (the manifest merging process will merge it only for debug builds):
   ```xml
   <manifest xmlns:android="http://schemas.android.com/apk/res/android"
       package="com.example.myapp">

       <!-- The INTERNET permission is required for development. Specifically,
            flutter needs it to communicate with the running application
            to allow setting breakpoints, to provide hot reload, etc.
       -->
       <uses-permission android:name="android.permission.INTERNET"/>

   </manifest>
   ```
1. Remove `<uses-permission android:name="android.permission.INTERNET"/>` from the main manifest
1. Update docs: https://flutter.io/android-release/#review-the-app-manifest

I tried the above and it works in VSCode and Android Studio.

Oh, and thank you so much for working on flutter! I'm still new, but everything fits so well together it's a lot of fun to work with. :)

**Comment by miyoyo** [comment]
I don't think this is a wise decision, since it means flutter apps will act differently from debug to release.

A better option might be to show a warning at compile time.

**Comment by zimmi** [comment]
I see your point, however that would still require editing the manifest for every release.
The cleanest solution would be to not require the internet permission for development, but that is probably out of scope.

If internet enabled apps are more common, another idea would be to have the permission both in debug and in main. The documentation should then make clear that the permission in main can be removed if it isn't needed.
The warning at compile time could still be added to find the mistake quicker.

**Comment by zoechi** [comment]
Looks similar to #8421

**Comment by jay5073** [comment]
this is not working with android pie version
